### PR TITLE
[FIX] Allow 0 to be passed as a config to print,write,plot,checkpoint_every and disable in that case

### PR DESCRIPTION
**Pull request by awennersteen**
This can be desired behavior in some cases, and is perhaps a better behaviour than the current where we end up raising DivideByZero errors.

We can perhaps use the `__post_int__` of the `TrainConfig` to do more validation here (like negative values), and print a warning?

Not sure what we think is best.

### CODE REVIEW

**Review by Roland-djee — COMMENTED** [review]
Thanks @awennersteen. Stupid question: does it make sense to default it to 0 and only check if it is non-negative ?

**Comment by awennersteen** [comment]
> Thanks @awennersteen. Stupid question: does it make sense to default it to 0 and only check if it is non-negative ?

I have no idea. My guess is that users usually will want these to be printed?

File: qadence/ml_tools/train_grad.py [hunk 1]
```diff
@@ -231,18 +225,22 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
...
                     print_metrics(loss, metrics, iteration - 1)
 
-                if iteration % config.write_every == 0:
+                if config.write_every > 0 and iteration % config.write_every == 0:
                     write_tracker(
                         writer, loss, metrics, iteration, tracking_tool=config.tracking_tool
                     )
 
-                if iteration % config.plot_every == 0:
+                if config.plot_every > 0 and iteration % config.plot_every == 0:
```
Comment (Roland-djee) [inline qadence/ml_tools/train_grad.py:243]: Is it possible to group them under a single `config.print_every > 0` check ?

File: qadence/ml_tools/train_no_grad.py [hunk 2]
```diff
@@ -114,13 +108,13 @@ def _update_parameters(
...
-            if iteration % config.print_every == 0 and config.verbose:
+            if config.print_every > 0 and iteration % config.print_every == 0 and config.verbose:
                 print_metrics(loss, metrics, iteration)
 
-            if iteration % config.write_every == 0:
+            if config.write_every > 0 and iteration % config.write_every == 0:
                 write_tracker(writer, loss, metrics, iteration, tracking_tool=config.tracking_tool)
 
-            if iteration % config.plot_every == 0:
+            if config.plot_every > 0 and iteration % config.plot_every == 0:
```
Comment (Roland-djee) [inline qadence/ml_tools/train_no_grad.py:117]: Same.

**Review by Roland-djee — APPROVED** [review]
Thanks @awennersteen. LGTM. Minor stylistic comments.

File: qadence/ml_tools/train_grad.py (hunk 1 above)
Comment (Roland-djee) [reply qadence/ml_tools/train_grad.py:243]: Ah yeah stupid me ofc.

**Pull request merged**